### ISSUE TITLE
DockerInstaller task permission issues on version 0.213.0

### ISSUE THREAD

**Issue by themaroqa** [microsoft/azure-pipelines-tasks#17352]
## Note  
It seems the PR #17131 introduced a bug

## Required Information

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DockerInstaller@0  

list here (V# not needed):  
https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/DockerInstallerV0

## Environment
- Azure Pipelines, hosted agent, os ubuntu-latest

## Issue Description

In the PR #17131 (so from version 0.213.0) the permissions have been modified, since that moment the DockerInstaller task is not able to detect the correct downloaded version: 
![image](https://user-images.githubusercontent.com/23500853/205889835-3fd9f9ad-92dd-4415-8aa1-800c38ed8059.png)

Later in the pipeline I tried to run a `docker -v` command and got the following error:
![image](https://user-images.githubusercontent.com/23500853/205891924-621a0d55-c52d-4454-867c-de55b755d7e6.png)

Pipeline example:
```
trigger: none

pool:
  vmImage: ubuntu-latest

steps:
- task: DockerInstaller@0
  inputs:
    dockerVersion: '19.03.8'

- task: PowerShell@2
  inputs:
    targetType: 'inline'
    script: 'docker -v'
```

**Comment by bluenote10** [comment]
Can confirm the issue, all our docker pipelines are broken as of today :confused:

**Comment by themaroqa** [comment]
@bluenote10 as a temp workaround you can fix the version of the task in the yml file
```
- task: DockerInstaller@0.209.0
  inputs:
    dockerVersion: '19.03.8'
```

**Comment by jedimanu-ux** [comment]
same issue with KubernetesV1

**Comment by themaroqa** [comment]
fix in progress #17442

**Comment by v-schhabra** [comment]
@jedimanu-ux can you please share the pipeline logs?

**Comment by sumisulai-js** [comment]
Was the fix released? The docker task still fails for 0.214.0

**Comment by v-schhabra** [comment]
The PR has been created and merged for this issue and mapped with 216 version.
https://github.com/microsoft/azure-pipelines-tasks/pull/17547/files
